dev

```diff
@@ -2,15 +2,18 @@
 
 set -uo pipefail
 
-this_dir=$(cd "$(dirname "$0")" && pwd)
-mkdir -p $this_dir/artifacts
-bazel build //pkg/cmd/dev --config nonogo &> $this_dir/artifacts/dev.log
-status=$?
-if [ $status -eq 0 ]
-then
-    $(bazel info bazel-bin --config nonogo)/pkg/cmd/dev/dev_/dev "$@"
-else
-    echo 'Failed to build pkg/cmd/dev! Got output:'
-    cat $this_dir/artifacts/dev.log
-    exit $status
+# Bump this counter (YYYYMMDD) to force rebuilding `dev` on all machines.
+DEV_VERSION=20220117
+
+THIS_DIR=$(cd "$(dirname "$0")" && pwd)
+BINARY_DIR=$THIS_DIR/bin/dev-versions
+BINARY_PATH=$BINARY_DIR/dev.$DEV_VERSION
+
+if [ ! -f "$BINARY_PATH" ]; then
+    echo "$BINARY_PATH not found, building..."
+    mkdir -p $BINARY_DIR
+    bazel build //pkg/cmd/dev --config nonogo
+    cp $(bazel info bazel-bin --config nonogo)/pkg/cmd/dev/dev_/dev $BINARY_PATH
```

Comment:
Nit: you can also stage a script in `bin/dev` of the form:

```
#!/usr/bin/env bash

$CRDBDIR/bin/dev-versions/dev.$DEV_VERSION
```

This would maintain compatibility with the previous versions which did stage `bin/dev`.